board.pl

```diff
@@ -1,3 +1,4 @@
+:- dynamic board_size/1.
```

Comment:
Suggestion for better naming:
horizontal_row -> row
vertical_row -> column
diagonal_row -> diagonal
What do you think?

Reply:
sure
and we can name the x,y,z like this as well if you want